### PR TITLE
#153 CcStrict validation works incorrectly

### PR DESCRIPTION
**Pull request by dmzaytsev**
task #153
Changed urn validation according with RFC 2141 in CcStrict, RFC 2141 compilant regexp took [here](https://github.com/jcabi/jcabi-urn/blob/master/src/main/java/com/jcabi/urn/URN.java#L98).
Also changed unit test

### CODE REVIEW

**Comment by davvd** [comment]
Let me find a reviewer for this pull request, thanks for submitting it

**Comment by davvd** [comment]
@pinaf review this one, please

File: src/test/java/org/takes/facets/auth/codecs/CcStrictTest.java [hunk 1]
```diff
@@ -67,5 +67,12 @@ public void passesValid() throws IOException {
                 )
             ), Matchers.equalTo("urn%3Atest%3A1")
         );
+        MatcherAssert.assertThat(
+            new String(
+                new CcStrict(new CcPlain()).encode(
+                    new Identity.Simple("urn:test:1:1")
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/codecs/CcStrictTest.java]: @dmzaytsev let's test with the urn provided in the original ticker: `urn:test-domain-org:valid:1`

File: src/main/java/org/takes/facets/auth/codecs/CcStrict.java [hunk 2]
```diff
@@ -67,7 +67,8 @@ private static Identity applyRules(final Identity identity) {
         if (identity.urn().isEmpty()) {
             throw new DecodingException("urn is empty");
         }
-        if (!identity.urn().matches("urn:[a-z]+:.+")) {
+        // @checkstyle LineLength (1 line)
+        if (!identity.urn().matches("^(?i)^urn(?-i):[a-z]{1,31}(:([\\-a-zA-Z0-9/]|%[0-9a-fA-F]{2})*)+(\\?\\w+(=([\\-a-zA-Z0-9/]|%[0-9a-fA-F]{2})*)?(&\\w+(=([\\-a-zA-Z0-9/]|%[0-9a-fA-F]{2})*)?)*)?\\*?$")) {
```
Comment (pinaf) [inline src/main/java/org/takes/facets/auth/codecs/CcStrict.java]: @dmzaytsev Wouldn't it be better to add a dependency to the `jcabi-urn` project and use method `boolean URN.isValid(String text)`?
@yegor256 what do you think?

**Comment by pinaf** [comment]
@dmzaytsev 2 comments above.

File: src/main/java/org/takes/facets/auth/codecs/CcStrict.java (hunk 2 above)
Comment (dmzaytsev) [inline src/main/java/org/takes/facets/auth/codecs/CcStrict.java]: @pinaf I'm not sure that we can use this dependence in the project

Comment (yegor256) [inline src/main/java/org/takes/facets/auth/codecs/CcStrict.java]: @pinaf we're trying to have zero dependencies in this framework

Comment (dmzaytsev) [inline src/main/java/org/takes/facets/auth/codecs/CcStrict.java]: @yegor256 @pinaf  Wouldn't it be better to add public static isUrnValid() to the class ?

**Comment by dmzaytsev** [comment]
@pinaf you're right regexp from jcabi-urn doesn't fully RFC compilant
according  [rfc2141](https://www.ietf.org/rfc/rfc2141.txt) 2.1 Namespace Identifier Syntax
```
<NID>         ::= <let-num> [ 1,31<let-num-hyp> ]
   <let-num-hyp> ::= <upper> | <lower> | <number> | "-"
   <let-num>     ::= <upper> | <lower> | <number>
```
I changed regexp as above

**Comment by pinaf** [comment]
@rultor merge
@dmzaytsev thanks

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, review this task for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@pinaf 

According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were only two issues found during review. Try to find at least three problems next time.

 According to our [quality rules](http://www.teamed.io/qa.html):
 > Messages in a ticket always start with a name of a user they are addressed to.

>davvd commented 8 days ago
Let me find a reviewer for this pull request, thanks for submitting it

And ask everyone to fix issues before ticket closing.

Please confirm that everything is clear and you have no questions regarding this approach.

**Comment by pinaf** [comment]
@elenavolokhova I confirm.

**Comment by elenavolokhova** [comment]
@davvd Ticket passed QA review. Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Ticket passed QA review. Quality is acceptable.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
@pinaf I added 10 mins to @elenavolokhova (for QA review) in transaction 55581921; done, I added **18 mins** in payment `AP-8KH556847R913132F`, 26 hours and 3 mins spent to complete this; extra minutes for review comments (c=3); added +18 to your rating, now it is equal to [+7172](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**